buddi/preprocessing/sc_preprocess.py

```diff
@@ -52,11 +50,11 @@ def get_cell_type_sum(in_adata, cell_type_id, num_samples):
   return sum_per_gene
 
 # method to generate a proportion vector
-def gen_prop_vec_lognormal(len_vector, num_cells):
+def gen_prop_vec_lognormal(len_vector, num_cells): # pseudo
 
   rand_vec = np.random.lognormal(5, np.random.uniform(1,3), len_vector) # 1
-
-  rand_vec = np.round((rand_vec/np.sum(rand_vec))*num_cells)
+  rand_vec = rand_vec + 1
+  rand_vec = np.ceil((rand_vec/np.sum(rand_vec))*num_cells)
```

Comment:
seems to have done the trick! I also noticed the 0 in the proportions when i used this function. after the rounding, some cells were 0. glad it's fixed:D

Reply:
yeah -- I wonder why I never caught this before -__-